File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
      Set <Character> hashset = new HashSet<Character> ();
      int i=0;
      int j=0;
      int maxLength=0;
      while(j<s.length())
      {
          if(!hashset.contains(s.charAt(j)))
          {
              hashset.add(s.charAt(j));
              // index=s.charAt(i);
              j++;
              maxLength=Math.max(maxLength,j-i);
              System.out.println(hashset);
              
          }
          else
          {
              hashset.remove(s.charAt(i));
              i++;
              
          }
       }
      
        return maxLength;
          
      
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        l=0;
        r=0;
        myset=set();
        maxlength =0;
        while r<len(s):
            if s[r] not in myset:
                myset.add(s[r]);
                maxlength = max(maxlength, r-l+1);
                r=r+1;
            else:
                myset.remove(s[l]);
                l=l+1;
        return maxlength;



        